Features/ManageabilityPkg/Universal/IpmiSmbiosTransferDxe/IpmiSmbiosTransferDxe.c

```diff
@@ -0,0 +1,293 @@
+
+  gBS->CloseEvent (Event);
+
+  Status = gBS->LocateProtocol (&gEdkiiIpmiBlobTransferProtocolGuid, NULL, (VOID **)&IpmiBlobTransfer);
+  if (EFI_ERROR (Status)) {
+    DEBUG ((DEBUG_ERROR, "%a: No IpmiBlobTransferProtocol available. Exiting\n", __func__));
+    goto ErrorExit;
+  }
+
+  SmbiosTransferRequired = TRUE;
+  Smbios30Table          = NULL;
+  Status                 = EfiGetSystemConfigurationTable (&gEfiSmbios3TableGuid, (VOID **)&Smbios30Table);
+  if (EFI_ERROR (Status) || (Smbios30Table == NULL)) {
+    DEBUG ((DEBUG_ERROR, "%a: No SMBIOS Table found: %r\n", __func__, Status));
+    REPORT_STATUS_CODE_WITH_EXTENDED_DATA (
+      EFI_ERROR_CODE | EFI_ERROR_MAJOR,
+      EFI_SOFTWARE_EFI_APPLICATION,
+      SMBIOS_EC_DESC_NO_SMBIOS_TABLE,
+      sizeof (SMBIOS_EC_DESC_NO_SMBIOS_TABLE)
+      );
+    return;
+  }
+
+  //
+  // BMC expects the Smbios Entry Point to point to the address within the binary data sent
+  // The value is initially pointing to the location in memory where the table lives
+  // So we will save off that value and then modify the entry point to make the BMC happy
+  Smbios30TableModified = AllocateZeroPool (sizeof (SMBIOS_TABLE_3_0_ENTRY_POINT));
+  CopyMem (Smbios30TableModified, Smbios30Table, sizeof (SMBIOS_TABLE_3_0_ENTRY_POINT));
+  Smbios30TableModified->TableAddress = sizeof (SMBIOS_TABLE_3_0_ENTRY_POINT);
+  //
+  // Fixup checksums in the Entry Point Structure
+  //
+  Smbios30TableModified->EntryPointStructureChecksum = 0;
+  Smbios30TableModified->EntryPointStructureChecksum =
+    CalculateCheckSum8 ((UINT8 *)Smbios30TableModified, Smbios30TableModified->EntryPointLength);
+
+  SendData = AllocateZeroPool (sizeof (SMBIOS_TABLE_3_0_ENTRY_POINT) + Smbios30Table->TableMaximumSize);
+  CopyMem (SendData, Smbios30TableModified, sizeof (SMBIOS_TABLE_3_0_ENTRY_POINT));
+  CopyMem (SendData + sizeof (SMBIOS_TABLE_3_0_ENTRY_POINT), (UINT8 *)Smbios30Table->TableAddress, Smbios30Table->TableMaximumSize);
+  SendDataSize      = sizeof (SMBIOS_TABLE_3_0_ENTRY_POINT) + Smbios30Table->TableMaximumSize;
+  RemainingDataSize = SendDataSize;
+
+  if (PcdGetBool (PcdSendSmbiosOnChanged)) {
+    SmbiosTransferRequired = DetectSmbiosChange (SendData, SendDataSize);
+
+    if (!SmbiosTransferRequired) {
+      DEBUG ((DEBUG_INFO, "%a: Smbios tables are not changed, skipping transfer to BMC\n", __func__));
+      return;
+    }
+  }
+
+  DEBUG_CODE_BEGIN ();
+  DEBUG ((SMBIOS_TRANSFER_DEBUG, "%a: SMBIOS BINARY DATA OUTPUT\n", __func__));
+  DEBUG ((SMBIOS_TRANSFER_DEBUG, "%a: Table Address: %x\n", __func__, Smbios30Table->TableAddress));
+  DEBUG ((SMBIOS_TRANSFER_DEBUG, "%a: Table Length: %x\n", __func__, Smbios30Table->TableMaximumSize));
+  for (Index = 0; Index < SendDataSize; Index++) {
+    if ((Index % BLOB_MAX_DATA_PER_PACKET) == 0) {
+      DEBUG ((SMBIOS_TRANSFER_DEBUG, "\nIndex:%x ", Index));
+    }
+
+    DEBUG ((SMBIOS_TRANSFER_DEBUG, "%02x ", *(SendData + Index)));
+  }
+

```

Comment:
```suggestion
  DEBUG ((SMBIOS_TRANSFER_DEBUG, "\n"));
```

New line required for the next message